app/scripts/controllers/network/fake-provider-engine.ts

```diff
@@ -0,0 +1,224 @@
+import { inspect, isDeepStrictEqual } from 'util';
```

Comment:
This file was mostly taken from a similar file in https://github.com/MetaMask/core/pull/1116. I say mostly because I ended up making some modifications to type things a little bit better and to remove some cruft that is not needed when using `json-rpc-engine` (for instance, there is no `stop` method on SafeEventEmitterProvider so that doesn't need to be overridden). Also I added a `beforeCompleting` property to the FakeProviderStub class as these tests needed it. I will copy these back to the core repo.